src/main.c

```diff
@@ -257,63 +257,53 @@ int main(int argc, char* args[])
   {
     char* defaultEggName = "game.egg";
     char* mainFileName = "main.wren";
-    char* fileName = defaultEggName;
 
     char* base = BASEPATH_get();
     char* arg = optparse_arg(&options);
-    if (arg != NULL) {
-      fileName = arg;
-    } else {
-      fileName = defaultEggName;
-    }
 
-    size_t baseLen = strlen(base);
-    size_t fileNameLen = strlen(fileName);
-    char* pathBuf = calloc(baseLen + fileNameLen + 1, sizeof(char));
-    strcpy(pathBuf, base);
-    strcat(pathBuf, fileName);
-    // Re-target basepath to the argument given
-    if (isDirectory(pathBuf)) {
-      BASEPATH_set(pathBuf);
-      fileName = defaultEggName;
-      fileNameLen = strlen(fileName);
-      arg = NULL;
-    } else {
-      BASEPATH_set(dirname(pathBuf));
-    }
-    base = BASEPATH_get();
-    baseLen = strlen(base);
+    char pathBuf[PATH_MAX];
 
-    if (isDirectory(pathBuf)) {
-      pathBuf = realloc(pathBuf, (baseLen + fileNameLen + 1) *  sizeof(char));
+    char* fileName = NULL;
+
+    if (arg != NULL) {
       strcpy(pathBuf, base);
-      strcat(pathBuf, defaultEggName);
+      strcat(pathBuf, arg);
+      if (isDirectory(pathBuf)) {
+        BASEPATH_set(pathBuf);
+      } else {
+        fileName = basename(pathBuf);
```

Comment:
Found the problem. `basename`/`dirname` can modify the string passed in, so we need to use a throwaway copy of `pathBuf` for each call here.

Reply:
Does 0c95335 solve the issue?

Reply:
Seems like it! Nice work!